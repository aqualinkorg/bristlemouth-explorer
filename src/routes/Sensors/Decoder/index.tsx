import { Button, Paper, Stack, Typography, styled } from '@mui/material';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';

const PaperContainer = styled(Paper)(({ theme }) => ({
  width: '20rem',
  height: '60vh',
  margin: '1rem 1rem 0.5rem 0.5rem',
  padding: '1rem',
  flexShrink: 0,
  borderRadius: theme.spacing(1),
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
}));

const StyledButton = styled(Button)(() => ({
  textTransform: 'none',
  borderRadius: '50px',
}));

const ButtonContentsWrapper = styled('div')(() => ({
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  width: '100%',
  paddingLeft: '1rem',
}));

const mockDecoderOptions = [
  'Decoder #1',
  'Decoder #2',
  'Decoder #3',
  'Create my own decoder',
];

function Decoder() {
  return (
    <PaperContainer>
      <Stack gap="0.5rem" textAlign="center" margin="1rem">
        <Typography variant="h5" fontWeight="bold">
          Select decoder
        </Typography>
        <Typography textAlign="left" fontSize="small">
          You can either select one of our decoders or create your own
        </Typography>
        <Stack gap="0.7rem">
          {mockDecoderOptions.map((option) => (
            <StyledButton key={option} variant="outlined">
              <ButtonContentsWrapper>
                <Typography>{option}</Typography>
                <ChevronRightIcon />
              </ButtonContentsWrapper>
            </StyledButton>
          ))}
        </Stack>
      </Stack>
    </PaperContainer>
  );
}

export default Decoder;
