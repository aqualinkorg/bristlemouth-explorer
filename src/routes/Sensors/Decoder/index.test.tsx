import { render } from '@testing-library/react';
import Decoder from '.';

it('renders as expected', () => {
  const { container } = render(<Decoder />);
  expect(container).toMatchSnapshot();
});
