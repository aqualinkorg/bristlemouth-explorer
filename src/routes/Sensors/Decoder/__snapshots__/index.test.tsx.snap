// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-cv4lt7-MuiPaper-root"
  >
    <div
      class="MuiStack-root css-1xxbk8x-MuiStack-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
      >
        Select decoder
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1q6e4rq-MuiTypography-root"
      >
        You can either select one of our decoders or create your own
      </p>
      <div
        class="MuiStack-root css-1l9ozal-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-19ku3c0-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <div
            class="css-juiud9"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Decoder #1
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-19ku3c0-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <div
            class="css-juiud9"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Decoder #2
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-19ku3c0-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <div
            class="css-juiud9"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Decoder #3
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-19ku3c0-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <div
            class="css-juiud9"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Create my own decoder
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
